src/App/Fossa/Analyze.hs

```diff
@@ -81,9 +71,62 @@ data ScanDestination
   = UploadScan URI ApiKey ProjectMetadata -- ^ upload to fossa with provided api key and base url
   | OutputStdout
 
-analyzeMain :: BaseDir -> Severity -> ScanDestination -> OverrideProject -> Bool -> IO ()
-analyzeMain basedir logSeverity destination project unpackArchives = withLogger logSeverity $
-  analyze basedir destination project unpackArchives
+analyzeMain :: BaseDir -> Severity -> ScanDestination -> OverrideProject -> Bool -> [BuildTargetFilter] -> IO ()
+analyzeMain basedir logSeverity destination project unpackArchives filters = withLogger logSeverity $
+  analyze basedir destination project unpackArchives filters
+
+discoverFuncs ::
+  ( Has (Lift IO) sig m,
+    MonadIO m,
+    Has ReadFS sig m,
+    Has Exec sig m,
+    Has Logger sig m,
+    Has Diag.Diagnostics sig m
+  ) =>
+  -- | Discover functions
+  [Path Abs Dir -> m [NewProject]]
+discoverFuncs =
+  [ Bundler.discover',
+    Cargo.discover',
+    Carthage.discover',
+    Cocoapods.discover',
+    Gradle.discover',
+    Rebar3.discover',
+    Gomodules.discover',
+    Godep.discover',
+    Setuptools.discover',
+    Maven.discover',
+    Leiningen.discover',
+    Composer.discover',
+    Cabal.discover',
+    Stack.discover',
+    Yarn.discover',
+    Npm.discover',
+    Scala.discover',
+    RPM.discover',
+    RepoManifest.discover'
+  ]
+
+runDependencyAnalysis ::
+  (Has (Lift IO) sig m, Has Logger sig m, Has (Output ProjectResult) sig m) =>
+  -- | Analysis base directory
+  BaseDir ->
+  [BuildTargetFilter] ->
+  NewProject ->
+  m ()
+runDependencyAnalysis (BaseDir basedir) filters project = do
+  case applyFiltersToProject basedir filters project of
+    Nothing -> logInfo $ "Skipping " <> pretty (projectType project) <> " project at " <> viaShow (projectPath project) <> ": no filters matched"
+    Just targets -> do
+      logInfo $ "Analyzing " <> pretty (projectType project) <> " project at " <> pretty (toFilePath (projectPath project))
+      graphResult <- sendIO . Diag.runDiagnosticsIO $ projectDependencyGraph project targets
+      Diag.withResult SevWarn graphResult (output . mkResult project)
+
+applyFiltersToProject :: Path Abs Dir -> [BuildTargetFilter] -> NewProject -> Maybe (Set BuildTarget)
+applyFiltersToProject basedir filters NewProject{..} =
+  case makeRelative basedir projectPath of
+    Nothing -> Just projectBuildTargets -- FIXME: this is required for --unpack-archives to continue to work
```

Comment:
What is this FIXME referring to?  What is wrong that needs to be fixed?

Reply:
Added more context here. Archives are unpacked to a temporary directory, so projectPath will be fail this `makeRelative`